#Q1. What are the three measures of central tendency?
Measures of central tendency help you find the middle, or the average, of a dataset. The 3 most common measures of central tendency are the mode, median, and mean.
(1)Mode: the most frequent value.t’s possible to have no mode, one mode, or more than one mode.The mode is most applicable to data from a nominal level of measurement.
(2)Median: the middle number in an ordered dataset.The median of a dataset is the value that’s exactly in the middle when it is ordered from low to high.
(3)Mean: the sum of all values divided by the total number of values.The arithmetic mean of a dataset  is the sum of all values divided by the total number of values.

#Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?

Measures of central tendency help you find the middle, or the average, of a dataset. The 3 most common measures of central tendency are the mode, median, and mean.
(1)Mode: the most frequent value.t’s possible to have no mode, one mode, or more than one mode.The mode is most applicable to data from a nominal level of measurement. Nominal data is classified into mutually exclusive categories, so the mode tells you the most popular category.To find the mode, sort your dataset numerically or categorically and select the response that occurs most frequently.
(2)Median: the middle number in an ordered dataset.The median of a dataset is the value that’s exactly in the middle when it is ordered from low to high.
(3)Mean: the sum of all values divided by the total number of values.The arithmetic mean of a dataset (which is different from the geometric mean) is the sum of all values divided by the total number of values. It’s the most commonly used measure of central tendency because all values are used in the calculation.

In [2]:
#Q3. Measure the three measures of central tendency for the given height data:
#[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
import numpy as np
from scipy import stats
l=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
print("mean:",np.mean(l))
print("median:",np.median(l))
print("mode:",stats.mode(l))

mean: 177.01875
median: 177.0
mode: ModeResult(mode=array([177.]), count=array([3]))


In [3]:
#Q4. Find the standard deviation for the given data:
#[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
l=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
print("standard deviation:",np.std(l))

standard deviation: 1.7885814036548633


#Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe
the spread of a dataset? Provide an example.
Measures of dispersion describe the spread of the data. They include the range, interquartile range, standard deviation and variance. 
(!).The range is given as the smallest and largest observations. This is the simplest measure of variability.ote in statistics (unlike physics) a range is given by two numbers, not the difference between the smallest and largest. For some data it is very useful, because one would want to know these numbers, for example knowing in a sample the ages of youngest and oldest participant. 
range of given observations: 32, 41, 28, 54, 35, 26, 23, 33, 38, 40.
Range (X) = Max (X) – Min (X)
= 54 – 23
= 31
(2).Quartiles and Interquartile Range
The quartiles, namely the lower quartile, the median and the upper quartile, divide the data into four equal parts; that is there will be approximately equal numbers of observations in the four sections (and exactly equal if the sample size is divisible by four and the measures are all distinct).
(3).Variance
The variance is a measure of variability that represents on how far each observation falls from the mean of the distribution.
Let’s calculate the variance of the follow data set: 2, 7, 3, 12, 9.
The squared differences for all values are added:
21.16 + 0.16 + 12.96 + 29.16 + 5.76 = 69.20
The sum is then divided by the number of data points:
69.20 ÷5 = 13.84
The variance is 13.84. To get the standard deviation, you calculate the square root of the variance, which is 3.72.
(4).Standard Deviation
However, one limitation to using the variance as our only measure of variability. When we square the numbers to get rid of the negatives (step 3), we also inadvertently square our unit of measurement. In other words, if we were talking about miles, we accidentally turned our unit of measurement into miles squared.for the below values [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172]
standard deviation: 1.7885814036548633

#Q6. What is a Venn diagram?
A Venn diagram is an illustration that uses circles to show the relationships among things or finite groups of things. Circles that overlap have a commonality while circles that do not overlap do not share those traits.
Venn diagrams help to visually represent the similarities and differences between two concepts. They have long been recognized for their usefulness as educational tools. Since the mid-20th century, Venn diagrams have been used as part of the introductory logic curriculum and in elementary-level educational plans around the world.
A Venn diagram uses circles that overlap or don't overlap to show the commonalities and differences among things or groups of things.
Things that have commonalities are shown as overlapping circles while things that are distinct stand alone.
Venn diagrams are now used as illustrations in business and in many academic fields.

#Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A∩B
here intersection of two set gives common elements.
Therefore A∩B =(2,6)
(ii) A ⋃ B
here Union of two sets give the set with all values and consider common values only once
Therefore, A ⋃ B=(2,3,4,5,6,7,0,8,10)

#Q8. What do you understand about skewness in data?
Skewness is a measurement of the distortion of symmetrical distribution or asymmetry in a data set. Skewness is demonstrated on a bell curve when data points are not distributed symmetrically to the left and right sides of the median on a bell curve. If the bell curve is shifted to the left or the right, it is said to be skewed.
1.Skewness, in statistics, is the degree of asymmetry observed in a probability distribution.
2.Distributions can exhibit right (positive) skewness or left (negative) skewness to varying degrees. A normal distribution (bell curve) exhibits zero skewness.
3.Investors note right-skewness when judging a return distribution because it, like excess kurtosis, better represents the extremes of the data set rather than focusing solely on the average.
4.Skewness informs users of the direction of outliers, though it does not tell users the number of outliers.

#Q9. If a data is right skewed then what will be the position of median with respect to mean?
The mean of positively skewed data will be greater than the median.When the distribution is skewed to the right, the mean is often greater than the median. In symmetric distributions, we expect the mean and median to be approximately equal in value. This is an important connection between the shape of the distribution and the relationship of the mean and median.
In case of a positively skewed frequency distribution, the mean is always greater than median and the median is always greater than the mode.
mean>median>mode is the relation of mean,median,mode of right skewed.

#Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?
Covariance 
1.Covariance is an indicator of the extent to which 2 random variables are dependent on each other. A higher number denotes higher dependency.
2.The value of covariance lies in the range of -∞ and +∞.
3.Change in scale Affects covariance
4.No Unit-free measure
5.The greater this number, the more reliant the relationship. Positive covariance denotes a direct relationship and is represented by a positive number.
A negative number, on the other hand, denotes negative covariance, which indicates an inverse relationship between the two variables. Covariance is great for defining the type of relationship, but it's terrible for interpreting the magnitude.
Correlation
1.correlationis a measure that determines the degree to which two or more random variables move in sequence. When an equivalent movement of another variable reciprocates the movement of one variable in some way or another during the study of two variables, the variables are said to be correlated. 
2.Correlation is limited to values between the range -1 and +1
3.Change in scale does not affect the correlation
4.Positive correlation occurs when two variables move in the same direction. When variables move in the opposite direction, they are said to be negatively correlated.

#Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.
Sample Mean = (Sum of terms) ÷ (Number of Terms)=∑xi/n
the sample mean of 60, 57, 109, 50
Sum of terms = 60 + 57 + 109 + 50 = 276
Number of terms = 4
Using sample mean formula,
mean = (sum of terms)/(number of terms)
mean = 276/4 = 69
Answer: The sample mean of 60, 57, 109, 50 is  69.

In [7]:
import seaborn as sns
import numpy as np
iris=sns.load_dataset('iris')
print('mean:',np.mean(iris['sepal_length']))

mean: 5.843333333333335


#Q12. For a normal distribution data what is the relationship between its measure of central tendency?
Any normal distribution has a graph that is perfectly symmetric about a vertical line through its peak. Therefore, all measures of central tendency (most commonly, the mean, median, and mode) give the same answer: the x-value of the peak.
The three common measures of central tendency of a distribution are the arithmetic mean, the median and the mode. Think of a distribution in terms of an histogram with many bars; a large sample from a normal distribution would describe a bell shaped curve that is symmetrical.
mean=median=mode are the relation of measure of central tendency for a normal distribution.The normal distribution is a bell-shaped, symmetrical distribution in which the mean, median and mode are all equal. If the mean, median and mode are unequal, the distribution will be either positively or negatively skewed

#Q13. How is covariance different from correlation?
Covariance 
1.Covariance is an indicator of the extent to which 2 random variables are dependent on each other. A higher number denotes higher dependency.
2.The value of covariance lies in the range of -∞ and +∞.
3.Change in scale Affects covariance
4.No Unit-free measure
Correlation
1.correlationis a measure that determines the degree to which two or more random variables move in sequence. When an equivalent movement of another variable reciprocates the movement of one variable in some way or another during the study of two variables, the variables are said to be correlated. 
2.Correlation is limited to values between the range -1 and +1
3.Change in scale does not affect the correlation

#Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.
Measures of central tendency are mean, median and mode.Outliers affect the mean value of the data but have little effect on
the median or mode of a given set of data.
Ten students take an exam and receive the following scores:
[0, 88, 90, 92, 94, 95, 95, 96, 97, 99]
The mean score is 84.6.
However, if we remove the “0” score from the dataset, then the mean score becomes 94.
Median: since the data set has 10 values, there are two middle
numbers, so one must find the mean of these two values, 85 and
90.
l=[0, 70, 70, 80, 85, 90, 90, 90, 95, 100]
87.5
Mode: The score 90 occurs more frequently than the other
values (three times), so 90 is the mode.
Receiving a zero on a quiz significantly affects a student’s mean,
or average. Notice that the outlier had a small effect on the
median and mode of the data.Receiving a zero on a quiz significantly affects a student’s mean,
or average. Notice that the outlier had a small effect on the
median and mode of the data.
It should be noted that because outliers affect the mean and have little effect on the median, the median is often used to describe “average” income. Often, one hears that the median income for a group is a certain value. Mean is not typically used because outliers, people who make significantly more or make no money at all, affect this measure. 